### PR TITLE
Move `benchmarks` subdomain links to `reports.risczero.com`

### PR DESCRIPTION
**Pull request by nuke-web3**
Just a fix of a nit on what is the "right" URL to use.

Closes https://linear.app/risczero/issue/WEBR0-23/update-all-links-to-use-reportsrisczerocom

Related to #2320

### CODE REVIEW

**Review by nahoc — APPROVED** [review]

**Review by pdg744 — APPROVED** [review]

**Comment by nuke-web3** [comment]
@pdg744 @nahoc - I can't merge this now, if you want it please do take it from here :pray:

**Comment by flaub** [comment]
Thanks, I think what we'll do is just update the links for next and new docs, but no need to adjust the existing links.

**Pull request closed without merging**
